### PR TITLE
Added order and order items entities. Implemented key endpoints.

### PR DESCRIPTION
**Pull request by DmytroV95**

### CODE REVIEW

File: src/main/java/com/varukha/onlinebookstore/mapper/OrderItemsMapper.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import com.varukha.onlinebookstore.config.MapperConfig;
+import com.varukha.onlinebookstore.dto.orderitem.OrderItemDto;
+import com.varukha.onlinebookstore.model.CartItem;
+import com.varukha.onlinebookstore.model.OrderItem;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(config = MapperConfig.class)
+public interface OrderItemsMapper {
+    @Mapping(target = "bookId", source = "orderItem.book.id")
```
Comment (irynamekh) [inline src/main/java/com/varukha/onlinebookstore/mapper/OrderItemsMapper.java:12]: You can simply write "book.id" instead "orderItem.book.id", result will be the same

**Review by irynamekh — APPROVED** [review]
Good job!

File: src/main/java/com/varukha/onlinebookstore/mapper/OrderItemsMapper.java (hunk 1 above)
Comment (andrii-hoienko) [reply src/main/java/com/varukha/onlinebookstore/mapper/OrderItemsMapper.java:12]: agree

File: src/main/java/com/varukha/onlinebookstore/mapper/OrderItemsMapper.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    OrderItemDto toDto(OrderItem orderItem);
+
+    @Mapping(target = "price", source = "cartItem.book.price")
```
Comment (andrii-hoienko) [inline src/main/java/com/varukha/onlinebookstore/mapper/OrderItemsMapper.java:15]: Same here

File: src/main/java/com/varukha/onlinebookstore/model/OrderItem.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    private int quantity;
+
+    @Column(nullable = false)
+    private BigDecimal price;
+
+    @Column(nullable = false)
+    private boolean isDeleted = false;
+
+    public OrderItem() {
```
Comment (andrii-hoienko) [inline src/main/java/com/varukha/onlinebookstore/model/OrderItem.java:45]: Please use lombok annotation  to create no args constructor

File: src/main/java/com/varukha/onlinebookstore/service/order/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+                .map(shoppingCart -> shoppingCart.getCartItems()
+                        .stream()
+                        .map(orderItemsMapper::toModel)
+                        .collect(Collectors.toSet()))
+                .orElse(Collections.emptySet());
+    }
+
+    @Transactional
+    public void clearShoppingCart(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
```
Comment (andrii-hoienko) [inline src/main/java/com/varukha/onlinebookstore/service/order/impl/OrderServiceImpl.java:127]: Why order service is responsible for interaction with shopping cart? Dont you have shopping cart service for it?

File: src/main/resources/db/changelog/changes/12-create-orders-item-table.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
+databaseChangeLog:
+  - changeSet:
+      id: create-order-item-table
+      author: Dmytro Varukha
```
Comment (andrii-hoienko) [inline src/main/resources/db/changelog/changes/12-create-orders-item-table.yaml:4]: add fk constraints

File: src/main/java/com/varukha/onlinebookstore/service/order/impl/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+    public OrderDto create(CreateOrderRequestDto orderRequestDto) {
+        User authenticatedUser = userService.getAuthenticatedUser();
+        Long userId = authenticatedUser.getId();
+        Set<OrderItem> orderItems = getOrderItems(userId);
+        Order order = buildOrder(authenticatedUser, orderRequestDto, orderItems);
+        clearShoppingCart(userId);
+        return orderMapper.toDto(order);
+    }
+
+    private Order buildOrder(User authenticatedUser,
```
Comment (andrii-hoienko) [inline src/main/java/com/varukha/onlinebookstore/service/order/impl/OrderServiceImpl.java:53]:
```suggestion
    private Order createOrder(User authenticatedUser,
```

File: src/main/java/com/varukha/onlinebookstore/service/order/impl/OrderServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+                        .map(orderItemsMapper::toDto)
+                        .collect(Collectors.toSet()))
+                .orElse(Collections.emptySet());
+    }
+
+    @Override
+    public OrderItemDto getOrderItemFromOrderById(Long orderId, Long itemId) {
+        return orderItemRepository.findOrderItemByIdAndOrderId(itemId, orderId)
+                .map(orderItemsMapper::toDto)
+                .orElseThrow();
```
Comment (andrii-hoienko) [inline src/main/java/com/varukha/onlinebookstore/service/order/impl/OrderServiceImpl.java:91]: @What is the purpose of empty orElseThrow?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
GJ, but some improvements required

**Review by andrii-hoienko — APPROVED** [review]
GJ!

**Pull request merged**